server/internal/infrastructure/gcs/file.go

```diff
@@ -104,6 +113,61 @@ func (f *fileRepo) RemoveAsset(ctx context.Context, u *url.URL) error {
 	return f.delete(ctx, sn)
 }
 
+func (f *fileRepo) UploadAssetFromURL(ctx context.Context, u *url.URL) (*url.URL, int64, error) {
+	if u == nil {
+		return nil, 0, errors.New("invalid URL")
+	}
+
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	req, err := http.NewRequestWithContext(ctxWithTimeout, http.MethodGet, u.String(), nil)
```

Comment:
Its not necessary that It will be on `GET`, Some might only Support `HEAD`. Please support more cases.

Reply:
@pyshx 
Thanks for your comment!

I think it would certainly increase the variety of http Method and some options, but I think most cases are `GET`.
On the other hand, I think some cases except `GET` are likely to be few and even if they are implemented, they are unlikely to be used.

It is my opinion that, in accordance with the `YAGNI` principles, there is no need to deal with many cases at the moment.

()

How do you think about it?